packages/manager/apps/web/client/app/hosting/database/private-sql-activation/private-sql-activation.routing.js

```diff
@@ -0,0 +1,58 @@
+import head from 'lodash/head';
+import map from 'lodash/map';
+import sortBy from 'lodash/sortBy';
+import toLower from 'lodash/toLower';
+
+export default /* @ngInject */ ($stateProvider) => {
+  $stateProvider.state('app.hosting.database.private-sql-activation', {
+    url: '/private-sql-activation',
+    component: 'hostingDatabasePrivateSqlActivation',
+    redirectTo: (transition) =>
+      Promise.all([
+        transition.injector().getAsync('Hosting'),
+        transition.injector().getAsync('HostingDatabase'),
+        transition.injector().getAsync('hosting'),
+      ]).then(([Hosting, HostingDatabase, hosting]) =>
+        HostingDatabase.getPrivateDatabaseCapabilities(hosting).then(
+          (capabilities) =>
+            Hosting.getPrivateDatabasesLinked(hosting).then((privateDbs) =>
+              privateDbs.length < capabilities.length
+                ? false
+                : { state: 'app.hosting' },
+            ),
+        ),
+      ),
+    resolve: {
+      me: /* @ngInject */ (User) => User.getUser(),
+      hosting: /* @ngInject */ ($transition$) =>
+        $transition$.params().productId,
+      services: /* @ngInject */ (OvhApiHostingWeb) =>
+        OvhApiHostingWeb.v6()
+          .query()
+          .$promise.then((services) =>
+            sortBy(services, (serviceName) => toLower(serviceName)),
+          ),
+      versions: /* @ngInject */ (PrivateDatabase, $translate) =>
+        PrivateDatabase.getAvailableOrderCapacities('classic')
+          .then(({ version }) =>
+            map(version, (v) => ({
+              id: v,
+              label: $translate.instant(
+                `privatesql_activation_version_v_${v.replace('.', '')}`,
+              ),
+            })),
+          )
+          .then((versions) => {
+            return sortBy(versions, [
+              ({ id }) => {
+                return head(id.split('_'));
+              },
+              ({ id }) => {
+                const [, version] = id.split('_');
+                return -1.0 * parseFloat(version);
```

Comment:
I understand the `parseFloat` but why the `-1.0` ? 😄

Reply:
it's to sort from highest to lowest (so the highest version is before in the list) i did not find a way to do a reverse order in the sortBy function

Reply:
@frenautvh `orderBy` with `'desc'` maybe ?

Reply:
FYI `-parseFloat(version)` would have been enough 🙂